packages/core/code-evaluation/rebuildGeometryCli.js

```diff
@@ -10,34 +10,27 @@ const { registerAllExtensions } = require('../io/registerExtensions')
 
 const rebuildSolids = (data) => {
   const defaults = { vtreeMode: false, serialize: false }
-  const { mainPath, vtreeMode, parameterValues, inputIsDirectory } = Object.assign({}, defaults, data)
+  let { mainPath, vtreeMode, parameterValues, inputIsDirectory } = Object.assign({}, defaults, data)
   const apiMainPath = vtreeMode ? '../code-loading/vtreeApi' : '@jscad/modeling'
   // we need to update the source for our module
   let requireFn = require
-  if (!inputIsDirectory) {
+
+  // source came from conversion, i.e. not from file system
+  if (data.source) {
```

Comment:
A few questions for clarity please:
- can we not reuse the inputIsDirectory in this case ? is there any difference ?
- why is mainPath overwritten ? ie does this get triggered *only* if there is a single input script ?

Reply:
I added the source versus prevsource to determine if the conversion took place. And adjusted rebuildGeometryCli() to use webRequire() if a source (converted source) is present.

The use of inputIsDirectory for single files was causing issues when *.stl, etc. files were passed in. For example, ./cli.js my.stl -of dxf

The use of data.source is not ideal. It would better be better to use a real option, i.e. isFromConversion.

For the webrequire to work properly, both fullPath and mainPath need to be the same. We could derive a 'in-memory' name from the original mainPath, but I don't see the reason.

Reply:
@kaosat-dev please review again.

i changed the logic slightly. there's a new option to rebuildGeometryCli called 'useFakeFs'. This option requestes rebuildGeometryCli to create a fake file system, and of course use webRequire as well.